aptos-move/framework/aptos-stdlib/sources/smart_vector.move

```diff
@@ -0,0 +1,416 @@
+module aptos_framework::smart_vector {
+    use std::error;
+    use std::vector;
+    use aptos_std::big_vector::{Self, BigVector};
+    use aptos_std::math64::max;
+    use aptos_std::type_info::size_of_val;
+
+    /// Vector index is out of bounds
+    const EINDEX_OUT_OF_BOUNDS: u64 = 1;
+    /// Vector is full
+    const EOUT_OF_CAPACITY: u64 = 2;
+    /// Cannot destroy a non-empty vector
+    const EVECTOR_NOT_EMPTY: u64 = 3;
+
+    /// A Scalable vector implementation based on tables, elements are grouped into buckets with `bucket_size`.
+    /// The internal vectors are actually option but it is intentional because Option requires `T` to be `drop` but
+    /// `SmartVector` should not enforece that.
+    struct SmartVector<T> has store {
+        inline_inner: vector<T>,
+        table_inner: vector<BigVector<T>>,
```

Comment:
why is this vector<BigVector<T>> not BigVector<T>?

Reply:
`vector` here is actually an `option` w/o `Drop` ability requirement of `T` to save space of a table handle when the smart_vector is small.

Reply:
can we have some documentation here? otherwise, it is quite hard to understand why having a vector here.